File: java/etl/src/main/java/Etl.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Etl {
    Map<String, Integer> transform(Map<Integer, List<String>> old) {
        HashMap<String, Integer> map = new HashMap<>();
        old.forEach((k, v) ->
                v.stream().forEach(x ->
                        map.put(x.toLowerCase(), k)));
        return map;
    }
}

File: cpp/etl/etl.cpp
#include "etl.h"
#include <ctype.h>


namespace etl
{

using std::map;
using std::vector;

const map<char, int> transform(const map<int, vector<char>>& old)
{
    map<char, int> ret;
    for (auto const& ent : old)
    {
        for (auto const& c : ent.second)
        {
            ret[tolower(c)] = ent.first;
        }
    }
    return ret;
}

}
